lexicons/app/bsky/embed/poll.json

```
@@ -0,0 +1,52 @@
+{
+  "lexicon": 1,
+  "id": "app.bsky.embed.poll",
+  "defs": {
+    "main": {
+      "type": "object",
+      "required": ["question", "options"],
+      "properties": {
+        "question": {
+          "type": "string",
+          "maxLength": 3000,
+          "maxGraphemes": 300,
+          "description": "The question being asked."
+        },
+        "options": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "maxLength": 3000,
+            "maxGraphemes": 300,
+            "description": "The options available for the poll."
+          },
+          "minLength": 2,
+          "maxLength": 4
+        }
```

This only covers single-choice polls.

Rather than a binary single-choice and any-number-of-choices system, it would be cool to have the maximum number of distinct selections given as an integer:

```suggestion
        },
        "maxAnswers": {
          "type": "integer",
          "minimum": 1,
          "default": 1
        }
```

It should be specified what happens when the number is reduced, but that's necessary anyway because multiple-choice polls can be edited into single-choice ones. (The easiest solution would be to discard/disregard all previous answers when poll post content is edited of course, which is what Mastodon does I believe.)

---

This is good to have too. I'm going to hold off on committing this in for now, as I would like to figure out the validation behavior that will allow this to work properly. Your mention of poll editing is key here. In the last question that I answered I was considering moving to having Polls be their own record, and, that may be necessary to handle poll editing properly, ensuring that PollAnswers are discarded, by nature of referencing a stale copy of the poll at edit time (that is to say, editing a poll would create a new poll record).